backend/controllers/org.controller.ts

```diff
@@ -45,3 +103,522 @@ export const createNewOrganization = async (
+        req.file!.originalname,
+        organizationId,
+      );
+      const path = `images/organizations/${uniqueFileName}`;
+      await UploadToS3(req.file!, path);
+      validatedUpdateOrganizationData.image = path;
+    }
+
+    // Update the organization
+    const updatedOrganization = await prisma.organization.update({
+      where: { id: organizationId },
+      data: {
+        ...validatedUpdateOrganizationData,
+      },
+    });
+    // send back the updated organization
+    if (updatedOrganization) {
+      // Organization updated successfully
+      res.status(200).json({
+        message: "Organization updated successfully",
+        data: updatedOrganization,
+      });
+    } else {
+      throw new AppError(
+        AppErrorName.EMPTY_RESULT_ERROR,
+        "Organization update returned empty result.",
+        500,
+        true,
+      );
+    }
+  } catch (error: any) {
+    next(error);
+  }
+};
+
+// Delete organization
+export const deleteOrganization = async (
+  req: RequestExtended,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    // Validate request id param
+    const organizationId = IdParamSchema.parse(req.params).id;
+
+    const loggedInUserId = req.userID;
+
+    // get the organization from the database
+    const existingOrganization = await prisma.organization.findUnique({
+      where: {
+        id: organizationId,
+      },
+    });
+
+    if (!existingOrganization) {
+      throw new AppError(
+        AppErrorName.NOT_FOUND_ERROR,
+        "Organization not found",
+        404,
+        true,
+      );
+    }
+
+    // check if the user has permission to delete the organization
+    const hasPermission = await checkUserPermission(
+      loggedInUserId!,
+      existingOrganization.id,
+      AppPermissionName.DELETE_ORGANIZATION,
+    );
+
+    if (!hasPermission) {
+      throw new AppError(
+        AppErrorName.PERMISSION_ERROR,
+        `User does not have permission to delete the organization`,
+        403,
+        true,
+      );
+    }
+
+    // Delete the organization
+    await prisma.organization.delete({
+      where: { id: organizationId },
+    });
+
+    res.status(204).end(); // No content after sucessful deletion
+  } catch (error: any) {
+    next(error);
+  }
+};
+
+// Get all Pending Organizations
+// Will be used by an admin dashboard
+export const getAllPendingOrganizations = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    // TODO: check the users system-level permissions (admin, not yet implemented)
+
+    // Fetch the roleId for the owner role
+    const { id: roleId } = await prisma.role.findUniqueOrThrow({
+      where: {
+        roleName: UserRole.Owner,
+      },
+      select: {
+        id: true,
+      },
+    });
+
+    const allOrgs = await prisma.organization.findMany({
+      where: {
+        status: OrganizationStatus.Pending,
+      },
+      // if we want to include owner information in the results:
```

Comment:
I think we should so that we have more information about whos trying to create an organization on our platform